File: app/src/main/java/com/prey/events/manager/EventMap.java
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2015 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.events.manager;

import java.util.HashMap;
import java.util.Iterator;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

@SuppressWarnings("serial")
public class EventMap<K, V> extends HashMap<K, V> {

    public JSONArray toJSONArray() {
        JSONArray jsonjArray = new JSONArray();
        Iterator<K> it = this.keySet().iterator();
        while (it.hasNext()) {
            K key = (K) it.next();
            JSONObject data = (JSONObject) this.get(key);
            jsonjArray.put(data);
        }
        return jsonjArray;
    }

    @SuppressWarnings("rawtypes")
    public JSONObject toJSONObject() {
        JSONObject jsonObject = new JSONObject();
        Iterator<K> it = this.keySet().iterator();
        while (it.hasNext()) {
            K key = (K) it.next();
            JSONObject data = (JSONObject) this.get(key);

            Iterator ite2 = data.keys();
            while (ite2.hasNext()) {
                String name = (String) ite2.next();
                try {
                    jsonObject.put(name, data.get(name));
                } catch (JSONException e) {
                }
            }
        }
        return jsonObject;
    }

    public boolean isCompleteData() {
        boolean isCompleteData = true;
        Iterator<K> it = this.keySet().iterator();
        while (it.hasNext()) {
            K key = (K) it.next();
            JSONObject data = (JSONObject) this.get(key);
            if (data == null) {
                isCompleteData = false;
                break;
            }
        }
        return isCompleteData;
    }

}

File: app/src/main/java/com/prey/events/manager/EventMap.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.events.manager

import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

/**
 * Checks if all values in the EventMap are non-null.
 *
 * @return True if all values are non-null, false otherwise.
 */
class EventMap<K, V> : HashMap<K, V>() {
    /**
     * Converts the EventMap to a JSONArray.
     *
     * @return JSONArray representation of the EventMap.
     */
    fun toJSONArray(): JSONArray {
        val jsonjArray = JSONArray()
        val it: Iterator<K> = keys.iterator()
        while (it.hasNext()) {
            val data = this[it.next()] as JSONObject?
            jsonjArray.put(data)
        }
        return jsonjArray
    }

    /**
     * Converts the EventMap to a JSONObject.
     *
     * @return JSONObject representation of the EventMap.
     */
    fun toJSONObject(): JSONObject {
        val jsonObject = JSONObject()
        val it: Iterator<K> = keys.iterator()
        while (it.hasNext()) {
            val data = this[it.next()] as JSONObject?
            val ite2: Iterator<*> = data!!.keys()
            while (ite2.hasNext()) {
                val name = ite2.next() as String
                try {
                    jsonObject.put(name, data[name])
                } catch (e: JSONException) {
                }
            }
        }
        return jsonObject
    }

    /**
     * Checks if all values in the EventMap are non-null.
     *
     * @return True if all values are non-null, false otherwise.
     */
    fun isCompleteData(): Boolean {
        var isCompleteData = true
        val it: Iterator<K> = keys.iterator()
        while (it.hasNext()) {
            val data = this[it.next()] as JSONObject?
            if (data == null) {
                isCompleteData = false
                break
            }
        }
        return isCompleteData
    }
}